platform/src/apis/Platform.Api.Insight/Expenditure/ExpenditureService.cs

```diff
@@ -67,6 +69,57 @@ public async Task<IEnumerable<SchoolExpenditureHistoryModel>> GetSchoolHistoryAs
         return await conn.QueryAsync<SchoolExpenditureHistoryModel>(sql, parameters);
     }
 
+    public async Task<IEnumerable<SchoolExpenditureHistoryModel>> GetSchoolHistoryAvgComparatorSetAsync(string urn, ExpenditureParameters queryParams)
+    {
+        var parameters = new
+        {
+            URN = urn
+        };
+
+        var sourceName = queryParams.Dimension switch
+        {
+            ExpenditureDimensions.Actuals => "SchoolExpenditureAvgComparatorSet",
+            ExpenditureDimensions.PerUnit => "SchoolExpenditureAvgPerUnitComparatorSet",
+            ExpenditureDimensions.PercentIncome => "SchoolExpenditureAvgPercentageOfIncomeComparatorSet",
+            ExpenditureDimensions.PercentExpenditure => "SchoolExpenditureAvgPercentageOfExpenditureComparatorSet",
+            _ => throw new ArgumentOutOfRangeException(nameof(queryParams))
+        };
+
+        var sql = $"SELECT * FROM {sourceName} WHERE URN = @URN";
+
+
+        using var conn = await dbFactory.GetConnection();
+        return await conn.QueryAsync<SchoolExpenditureHistoryModel>(sql, parameters);
+    }
+
+    public async Task<IEnumerable<SchoolExpenditureHistoryModel>> GetSchoolHistoryAvgNationalAsync(ExpenditureParameters queryParams)
+    {
+        if (string.IsNullOrEmpty(queryParams.OverallPhase) || string.IsNullOrEmpty(queryParams.FinanceType))
```

Comment:
Please can this be done in the parameters validator?

Reply:
added new validation and parameters for the national average endpoint. 

This seems better I think, the existing parameters/validation are untouched and reused for the comparator set endpoint.

Let me know what you think!